packages/flutter_test/lib/src/window.dart

```diff
@@ -947,6 +973,122 @@ class TestFlutterView implements FlutterView {
+  ///   * [reset] to reset all test values for this display
+  @override
+  double get devicePixelRatio => _devicePixelRatio ?? _display.devicePixelRatio;
+  double? _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    _devicePixelRatio = value;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [devicePixelRatio] to the default value for this display.
+  ///
+  /// This will also reset the [devicePixelRatio] for any [TestFlutterView]s
+  /// that are related to this display.
+  void resetDevicePixelRatio() {
+    _devicePixelRatio = null;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// The refresh rate to use for this test.
+  ///
+  /// Defaults to the value provided by [Display.refreshRate]. This
+  /// can only be set in a test environment to emulate different display
+  /// configurations. A standard [Display] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [Display.refreshRate] for the standard implementation
+  ///   * [resetRefreshRate] to reset this value specifically
+  ///   * [reset] to reset all test values for this display
+  @override
+  double get refreshRate => _refreshRate ?? _display.refreshRate;
+  double? _refreshRate;
+  set refreshRate(double value) {
+    _refreshRate = value;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [refreshRate] to the default value for this display.
+  void resetRefreshRate() {
+    _refreshRate = null;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// The size of the [Display] to use for this test.
+  ///
+  /// Defaults to the value provided by [Display.refreshRate]. This
+  /// can only be set in a test environment to emulate different display
+  /// configurations. A standard [Display] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [Display.refreshRate] for the standard implementation
+  ///   * [resetRefreshRate] to reset this value specifically
+  ///   * [reset] to reset all test values for this display
+  @override
+  Size get size => _size ?? _display.size;
+  Size? _size;
+  set size(Size value) {
+    _size = value;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [size] to the default value for this display.
+  void resetSize() {
+    _size = null;
+    _platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets all values on this [TestDisplay].
+  ///
+  /// See also:
+  ///   * [resetDevicePixelRatio] to reset [devicePixelRatio] specifically
+  ///   * [resetRefreshRate] to reset [refreshRate] specifically
+  ///   * [resetSize] to reset [size] specifically
+  void reset() {
+    resetDevicePixelRatio();
+    resetRefreshRate();
+    resetSize();
+  }
+
+  /// This gives us some grace time when the dart:ui side adds something to
+  /// [Display], and makes things easier when we do rolls to give
+  /// us time to catch up.
+  @override
+  dynamic noSuchMethod(Invocation invocation) {
+    return null;
+  }
```

Comment:
nit: consider `extends Fake` instead of implmeenting nSM yourself.

Reply:
I'm going to hold off on making this change here for now. The other `Test*` classes in this file make use of a `noSuchMethod` implementation that returns null and I'd like to investigate the impact of changing that separately from this ticket.

I do like the idea though and would like to apply it to _all_ the `Test*` classes at the same time if possible, so I wrote up #127613 to keep track of it!